data-prepper-plugins/kinesis-source/src/main/java/org/opensearch/dataprepper/plugins/kinesis/source/processor/KinesisRecordProcessor.java

```diff
@@ -161,16 +170,21 @@ public void processRecords(ProcessRecordsInput processRecordsInput) {
 
             // Track the records for checkpoint purpose
             kinesisCheckpointerTracker.addRecordForCheckpoint(extendedSequenceNumber, processRecordsInput.checkpointer());
-            List<Record<Event>> records = kinesisRecordConverter.convert(
+
+            List<KinesisInputOutputRecord> kinesisOutputRecords = kinesisRecordConverter.convert(
                     kinesisStreamConfig.getCompression().getDecompressionEngine(),
                     processRecordsInput.records(), streamIdentifier.streamName());
 
             int eventCount = 0;
-            for (Record<Event> record: records) {
-                Event event = record.getData();
+            for (KinesisInputOutputRecord kinesisInputOutputRecord: kinesisOutputRecords) {
+                Record<Event> dataPrepperRecord = kinesisInputOutputRecord.getDataPrepperRecord();
+                int incomingRecordSizeBytes = kinesisInputOutputRecord.getKinesisClientRecord().data().position();
```

Comment:
Can you use ByteBuffer's `capacity` instead?

https://docs.oracle.com/javase/8/docs/api/java/nio/Buffer.html#capacity--

Per my comment above, you could calculate this elsewhere.

Reply:
@dlvenable - It looks like `capacity` is the total size of the buffer. To get KCL record bytes, it might not be using the entire capacity. However, `position` returns the offset upto which the data has been read.